core/modules/meet/client/meet.js

```diff
@@ -63,11 +67,21 @@ const onZoneLeft = e => {
 const onZoneUpdated = e => {
   if (!linkedZoneId) return;
 
+  const meetingRoomService = meetingRoom.getMeetingRoomService();
+  const linkedZone = Zones.findOne(linkedZoneId);
```

Comment:
I think you can move this logic further down and use `const { zone } = e.detail;` which contains the info about the zone that was just updated.

Using `linkedZoneId` might cause a bug if you are in a meeting room and another zone is updated.

Reply:
I think it's the opposite.
It seems to me that the `linkedZoneId` corresponds to the last zone you enter/exit, that's the only moment the reference is modified.
Whereas `const { zone } = e.detail;` corresponds to an update of a zone, not necessarily the one you are currently in.
If I use `const { zone } = e.detail;`, I have to compare its ID to `linkedZoneId` anyway to be sure that the updated zone is the current one, unless I manually check if the user is in the zone with the coordinates.

Therefore, I am not sure whether I understood the comment correctly

Reply:
My bad, I misread your first version. In fact, the find was not necessary I think. You have to check that the updated area has no name anymore and if it corresponds to the current meeting area.

```
if (!zone.roomName && zone._id === linkedZoneId)
```

Or maybe I missed something  😅

Reply:
Well, I believe this is another way to implement it.

Since you must be in a linkedZone to reach this statement, checking if the current zone (linkedZone) has a roomname is the same as checking `if (!zone.roomName && zone._id === linkedZoneId)`

I'll have a look next week and can surely change it for the sake of readability, but I think it does not change anything.